test.el

```diff
@@ -34,6 +34,15 @@
 (ert-deftest ffi-add ()
   (should (eq (test-add 23 -23) 0)))
 
+(defun callback-with-ffi-add-call (arg)
+	(test-add 1 arg)
+	)
+
+(ert-deftest ffi-call-callback-with-ffi-add-call ()
+  (let* ((cif (ffi--prep-cif :int [:int]))
+	 (pointer-to-callback (ffi-make-closure cif #'callback-with-ffi-add-call)))
+    (should (eq (test-call-callback pointer-to-callback) "hello"))))
```

Comment:
Something is strange here, because `test-call-callback` is going to return an integer, but this test is checking to see if it is `eq` to a string -- which can't happen.

When I change this to expect 23, it works locally.

I wouldn't doubt that there is a problem -- that issue you pointed at has never been fixed -- but I think some other test case is required to show it.